src/pages/core/guides/anchor/anchor-staking-example.md

```diff
@@ -0,0 +1,481 @@
+
+_Learn more about it [here](/core/plugins/attribute)_
+
+### The Smart Contract Logic
+
+For simplicity, this example includes only two instructions: the **stake** and **unstake** functions since are essential for a staking program to work as intended. While additional instructions, such as a **spendPoint** instruction, could be added to utilize accumulated points, this is left to the reader to implement. 
+
+_Both the Stake and Unstake functions utilize, differently, the plugins introduced previously_.
+
+Before diving into the instructions, let’s spend some time talking about the attributes used, the `staked` and `staked_time` keys. The `staked` key indicates if the asset is staked and when it was staked if it was (unstaked = 0, staked = time of staked). The `staked_time` key tracks the total staking duration of the asset, updated only after an asset get’s unstaked.
+
+**Instructions**:
+- **Stake**: This instruction applies the Freeze Delegate Plugin to freeze the asset by setting the flag to true. Additionally, it updates the`staked` key in the Attribute Plugin from 0 to the current time.
+- **Unstake**: This instruction changes the flag of the Freeze Delegate Plugin and revokes it to prevent malicious entities from controlling the asset and demanding ransom to unfreeze it. It also updates the `staked` key to 0 and sets the `staked_time` to the current time minus the staked timestamp.
+
+## Building the Smart Contract: A Step-by-Step Guide
+
+Now that we understand the logic behind our smart contract, **it’s time to dive into the code and bring everything together**!
+
+### Anchor Overview
+
+In this guide, we’ll use the Anchor framework, but you can also implement it using a native program. Learn more about the Anchor framework here: [Anchor Framework](https://book.anchor-lang.com/introduction/what_is_anchor.html).
+
+For simplicity and the sake of this exercise, we’ll use a mono-file approach with accounts and instructions all in lib.rs instead of the usual separation.
+
+**Note**: You can follow along and open the example in Solana Playground, an online tool to build and deploy Solana programs: Solana Playground.
+
+In the account struct of all instructions, we will separate the Signer and the Payer. This is a standard procedure because PDAs cannot pay for account creation, so if the user wants a PDA to be the authority over the instruction, there need to be two different fields for it. While this separation isn't strictly necessary for our instructions, it's considered good practice.
+
+### The Account Struct
+
+For this example we used the anchor flag from the mpl-core crate to directly deserialize the Asset and Collection account from the account struct and put some constraint on that
+
+_Learn more about it [here](/core/using-core-in-anchor)_
+
+The account struct for both the `Stake` and `Unstake` instruction are the same since we're going to use the same account with the same constraint for both 
+
+```rust
```

Comment:
there are two lines in here that make me scroll horizonally. Can we format it differently?

Reply:
This i don't think i could format them differently. The only thing i'm thinking is 
```rust
constraint = asset.update_authority == 
UpdateAuthority::Collection(collection.key()),
```

And i don't think is great

Reply:
this won't be an issue when the larger layout goes live.